test/e2e/common_suite_test.go

```diff
@@ -114,15 +112,58 @@ func (tc *testCase) run() {
 				if err = client.Resources().Create(ctx, tc.pod); err != nil {
 					t.Fatal(err)
 				}
-				if err = wait.For(conditions.New(client.Resources()).PodRunning(tc.pod), wait.WithTimeout(WAIT_POD_RUNNING_TIMEOUT)); err != nil {
-					t.Fatal(err)
+				if tc.expectedPodLogString != "" {
+					if err = wait.For(conditions.New(client.Resources()).PodPhaseMatch(tc.pod, v1.PodSucceeded), wait.WithTimeout(WAIT_POD_RUNNING_TIMEOUT)); err != nil {
+						t.Fatal(err)
+					}
+				} else {
+					if err = wait.For(conditions.New(client.Resources()).PodRunning(tc.pod), wait.WithTimeout(WAIT_POD_RUNNING_TIMEOUT)); err != nil {
+						t.Fatal(err)
+					}
 				}
+
 			}
 			return ctx
 		}).
 		Assess(tc.assessMessage, func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
 			client := cfg.Client()
 			var podlist v1.PodList
+			if tc.pod != nil {
```

Comment:
Any particularly reason that you moved the `if tc.pod != nil` logic up above the `if tc.job != nil` branch in this PR?

Reply:
No It wont affect the logic of the code.  Placement can be changed if required

Reply:
Yes - I'd rather keep the original placement as it makes the diff in the PR smaller, thanks

Reply:
I think we are still waiting for this fix?

Reply:
Exchanged `job` and `pod` statements in updated code